bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpUtil.java

```diff
@@ -377,7 +379,22 @@ public static RawType downloadImage(String url) {
      *         not an image or the data size is too big
      */
     public static RawType downloadImage(String url, boolean scanTypeInContent, long maxContentLength) {
-        return downloadData(url, "image/.*", scanTypeInContent, maxContentLength);
+        return downloadImage(url, scanTypeInContent, maxContentLength, DEFAULT_TIMEOUT_MS);
+    }
+
+    /**
+     * Download the image data from an URL.
+     *
+     * @param url the URL of the image to be downloaded
+     * @param scanTypeInContent true to allow the scan of data to determine the content type if not found in the headers
+     * @param maxContentLength the maximum data size in bytes to trigger the download; any negative value to ignore the
+     *            data size
+     * @param timeout the socket timeout in milliseconds to wait for data
+     * @return a RawType object containing the image, null if the content type could not be found or the content type is
+     *         not an image or the data size is too big
+     */
+    public static RawType downloadImage(String url, boolean scanTypeInContent, long maxContentLength, int timeout) {
```

Comment:
Given the current executeurl methods putting the timeout in the end is already so maybe just do it, it makes the code In your binding better.

Reply:
Yes, that is what is done, the timeout parameter is the last parameter.
Sorry I don't understand your comment.

Reply:
I referred to my own earlier comment:

> It might make the API a little bit confusing if the second parameter is sometimes a boolean scanTypeInContent and sometime the timeout.

I yesterday noticed (because it was shown in the diff now) that the existing methods have the same pattern applied that every method has the timeout as the last parameter, meaning that it is sometimes the second, third or fourth.

So I was actually down voting my previous comment and suggesting that if it was needed for better code in the calling side, that it is proper to append it also to the single argument method.